backend/app.py

```diff
@@ -308,6 +315,178 @@ def create_machine():
+    except ValueError:
+        # just put the multimachine as a string, we need to correct it then
+        multimachine = "TODO" + str(request.args.get("multimachine"))
+    if multimachine != 1:
+        updated_machine_entry["properties"]["multimachine"] = multimachine
+        change_message += " multimachine,"
+
+    # Case 5: paywall reported
+    paywall_new = True if request.args.get("paywall") == "true" else False
+    paywall_old = existing_machine_infos["properties"].get("paywall", False)
+    if paywall_new != paywall_old:
+        updated_machine_entry["properties"]["paywall"] = paywall_new
+        change_message += " paywall,"
+
+    # Case 6: address and / or location changed --> check for their correspondence
+    (lng_old, lat_old) = existing_machine_infos["geometry"]["coordinates"]
+    old_address = existing_machine_infos["properties"]["address"]
+    # if address or coordinates were changed, compare them and return warning if needed
+    address_okay = True
+    if latitude != lat_old or longitude != lng_old or address != old_address:
+        # Verify that address matches coordinates
+        found_coords, (lat, lng) = verify_machine_location(address, area, title)
+        # if address was changed but is not found (error only if address was changed)
+        if (not found_coords) and address != old_address:
+            return jsonify({"error": "Google Maps does not know this address"}), 400
+
+        dist = haversine((lat, lng), (latitude, longitude))
+        if dist > 1:  # km
+            address_okay = False  # triggers warning
+
+        # adapt dictionary entries
+        updated_machine_entry["properties"]["address"] = address
+        updated_machine_entry["properties"]["latitude"] = str(latitude)
+        updated_machine_entry["properties"]["longitude"] = str(longitude)
+        updated_machine_entry["geometry"]["coordinates"] = [longitude, latitude]
+        if address != old_address:
+            change_message += " address,"
+        if latitude != lat_old or longitude != lng_old:
+            change_message += " location,"
+
+    # process machine change in thread
+    thread = Thread(
+        target=process_machine_change,
+        args=(updated_machine_entry, request.remote_addr, change_message),
+    )
+    thread.start()
+
+    # return warning if the address and coordinates do not correspond
+    if not address_okay:
+        return (
+            jsonify(
+                {
+                    "error": f"Change request submitted successfully. However, the address ({address}) seems >1km away from coordinates ({latitude}, {longitude}). Consider adjusting your edits such that coordinates and address are aligned."
+                }
+            ),
+            300,
+        )
+    return jsonify({"message": "Success!"}), 200
+
+
+def process_machine_change(
+    updated_machine_entry: dict,
+    ip_address: str,
+    change_message: str,
+    wait_buffer_min: int = 5,
+):
```

Comment:
Docstring missing.
Also, this function might better belong to `github_update.py`

Reply:
Done